addons/arsenal/XEH_preInit.sqf

```diff
@@ -15,55 +15,61 @@ PREP_RECOMPILE_END;
 [QGVAR(statsToggle), {
     params ["_display", "_showStats"];
 
+    _showStats = GVAR(showStats) && _showStats;
+
     private _statsCtrlGroupCtrl = _display displayCtrl IDC_statsBox;
     private _statsPreviousPageCtrl = _display displayCtrl IDC_statsPreviousPage;
     private _statsNextPageCtrl = _display displayCtrl IDC_statsNextPage;
     private _statsCurrentPageCtrl = _display displayCtrl IDC_statsCurrentPage;
 
     {
-        _x ctrlShow (GVAR(showStats) && {_showStats});
+        _x ctrlShow _showStats;
     } forEach [
         _statsCtrlGroupCtrl,
         _statsPreviousPageCtrl,
         _statsNextPageCtrl,
         _statsCurrentPageCtrl
     ];
-}] call CBA_fnc_addEventHandler;
 
-[QGVAR(statsChangePage), {
-    _this call FUNC(buttonStatsPage);
-}] call CBA_fnc_addEventHandler;
+    if (!_showStats) exitWith {};
 
-[QGVAR(displayStats), {
-    _this call FUNC(handleStats);
-}] call CBA_fnc_addEventHandler;
+    GVAR(statsInfo) params ["", "_control", "_curSel", "_itemCfg"];
 
-[QGVAR(actionsChangePage), {
-    _this call FUNC(buttonActionsPage);
+    [QGVAR(displayStats), [_display, _control, _curSel, _itemCfg]] call CBA_fnc_localEvent;
 }] call CBA_fnc_addEventHandler;
 
-[QGVAR(displayActions), {
-    _this call FUNC(handleActions);
-}] call CBA_fnc_addEventHandler;
+[QGVAR(statsChangePage), LINKFUNC(buttonStatsPage)] call CBA_fnc_addEventHandler;
+[QGVAR(displayStats), LINKFUNC(handleStats)] call CBA_fnc_addEventHandler;
 
 [QGVAR(actionsToggle), {
     params ["_display", "_showActions"];
 
+    _showActions = GVAR(showActions) && _showActions;
+
     private _actionsCtrlGroupCtrl = _display displayCtrl IDC_actionsBox;
     private _actionsPreviousPageCtrl = _display displayCtrl IDC_actionsPreviousPage;
     private _actionsNextPageCtrl = _display displayCtrl IDC_actionsNextPage;
     private _actionsCurrentPageCtrl = _display displayCtrl IDC_actionsCurrentPage;
 
     {
-        _x ctrlShow (GVAR(showActions) && {_showActions});
+        _x ctrlShow _showActions;
     } forEach [
         _actionsCtrlGroupCtrl,
         _actionsPreviousPageCtrl,
         _actionsNextPageCtrl,
         _actionsCurrentPageCtrl
     ];
+
+    if (!_showActions) exitWith {};
+
+    GVAR(actionsInfo) params ["_control", "_curSel", "_itemCfg"];
+
+    [QGVAR(displayActions), [_display, _control, _curSel, _itemCfg]] call CBA_fnc_localEvent;
```

Comment:
Not a huge fan of this solution, as I feel it could fail.

Reply:
How so? Isn't this just doing the same thing as selecting a different item/switching tabs would?

Reply:
I'm wondering: If the arsenal refreshes and removes the currently selected item, what happens?